### PR TITLE
refactor: address and resolve TODOs

### LINKED ISSUE #17342: Review and address TODOs
Review and address all the TODOs into the project, refactor the code where needed and improve maintainability

Blocked by #17340

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

- made job maxRetries configurable via @JobWorker annotation
- moved metrics to command callback
- changed the way how the SDK handle error code, throwing an error every also when an endpoint is not found
- removed old todos

## Related issues

closes #17342

### CODE REVIEW

File: spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/annotation/JobWorker.java [hunk 1]
```diff
@@ -105,4 +105,7 @@
 
   /** Stream timeout in ms */
   long streamTimeout() default -1L;
+
+  /** Set the max number of retries for a job */
+  int maxRetries() default 5;
```
Comment (ana-vinogradova-camunda) [inline spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/annotation/JobWorker.java:110]: 💭 question for my own understanding: is 5 a usual default for the retries?

**Review by ana-vinogradova-camunda — APPROVED** [review]
Thanks for a great job, LGTM 👍 
Just one question, but nothing major.

File: spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/annotation/JobWorker.java (hunk 1 above)
Comment (nicpuppa) [reply spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/annotation/JobWorker.java:110]: Not really, do you think is better to have `-1` as default value ?

Comment (ana-vinogradova-camunda) [reply spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/annotation/JobWorker.java:110]: 💭 `-1` looks to be consistent with the rest of definitions, we might want to follow that

**Pull request merged**